Source/IO/SmallDataIO.cpp

```diff
@@ -254,10 +271,437 @@ void SmallDataIO::remove_duplicate_time_data(const bool keep_m_time_data)
 
 // ------------ Reading Functions ------------
 
+void SmallDataIO::file_structure_t::clear()
+{
+    num_blocks = 0;
+    block_starts.clear();
+    num_data_rows.clear();
+    num_header_rows.clear();
+    num_data_columns.clear();
+}
+
+// Only for use in the constructor - this assumes that the file is already open
+void SmallDataIO::read_file()
+{
+
+    AMREX_ASSERT_WITH_MESSAGE(
+        m_file, "This function can only be called from the constructor! \n");
+
+    m_file.seekg(0, std::ios::end);
+
+    auto size = m_file.tellg();
+
+    AMREX_ASSERT_WITH_MESSAGE(size < max_file_size,
+                              "File size is too large for SmallDataIO.\n");
```

Comment:
Do you want to make this an `AMREX_ALWAYS_ASSERT_WITH_MESSAGE` so it's triggered even if not compiling with `DEBUG = TRUE`?

Also, I think it would be nice to prepend error messages with this class name:
```suggestion
    AMREX_ASSERT_WITH_MESSAGE(size < max_file_size,
                              "SmallDataIO: File size is too large.\n");
```